### ISSUE TITLE
NeDB Support

### ISSUE THREAD

**Issue by mindjuice** [JKHeadley/rest-hapi#170]
**Is your feature request related to a problem? Please describe.**
For smaller projects where a full mongodb server is overkill, it would be great to use `rest-hapi` with NeDB: https://github.com/louischatriot/nedb

**Describe the solution you'd like**
Rather than providing a mongo URI, we could pass in an object that maps model name keys to NeDB datastore instances (i.e., mongo collections).

**Describe alternatives you've considered**
TingoDB, but it appears to be abandoned.

**Additional context**
I haven't looked in detail yet to see if `rest-hapi` uses any parts of the mongo API that NeDB does not support.

If this seems possible, I can help with the code changes.

**Comment by mindjuice** [comment]
Looks like this might already be possible by passing in a mongoose instance to the options object that is wrapped by NeDB:

https://github.com/aerys/mongoose-nedb
https://github.com/aerys/mongoose-nedb/blob/master/lib/connection.js
https://github.com/aerys/mongoose-nedb/blob/master/lib/collection.js

That package does not seem active/supported though.

**Comment by JKHeadley** [comment]
I love this idea! It does look like NeDB isn't very actively supported, though it might be worth looking into. 

Another possible alternative if you just want an in-memory db is https://github.com/nodkz/mongodb-memory-server, though it is mostly meant for testing purposes.

You can see it in use in the rest-hapi e2e tests: https://github.com/JKHeadley/rest-hapi/blob/b069959a8813ecc01124fe38e8b03817ff66b6b9/tests/e2e/end-to-end.tests.js#L12

**Comment by mindjuice** [comment]
For my use case, I need persistence (it's for a small, Raspberry-Pi-like device), and data would be streaming in quite frequently.  If I can get 2GB of RAM onto the unit I might be confident enough to just use MongoDB directly.  Worried about the overhead of MongoDB on one hand, and whether NeDB would be up to the task on the other hand.

**Comment by JKHeadley** [comment]
Yeah that makes total sense. Just curious, if you could choose your ideal DB for your application, what would it be? I would like to one day update rest-hapi to support multiple DBs.

**Comment by mindjuice** [comment]
I've mainly used MongoDB and MariaDB/MySQL, but I also really liked RethinkDB before the company went under.  I sort of lost hope for it, but it looks like the community has kept it up pretty well though.

It has the ability to stream updates to queries in real-time as the database changes, which is really slick.
